launchdarkly/ai_model_config_helper.go

```diff
@@ -0,0 +1,185 @@
+			ValidateDiagFunc: validateKey(),
+		},
+		KEY: {
+			Type:             schema.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			ValidateDiagFunc: validateKey(),
+			Description:      addForceNewDescription("The unique key that references the AI model config. A change in this field will force the destruction of the existing resource and the creation of a new one.", !isDataSource),
+		},
+		NAME: {
+			Type:        schema.TypeString,
+			Required:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("The human-friendly name for the AI model config.", !isDataSource),
+		},
+		ID: {
+			Type:        schema.TypeString,
+			Required:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Identifier for the model, for use with third party providers.", !isDataSource),
+		},
+		PROVIDER: {
+			Type:        schema.TypeString,
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Provider for the model.", !isDataSource),
+		},
+		ICON: {
+			Type:        schema.TypeString,
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Icon for the model.", !isDataSource),
+		},
+		PARAMS: {
+			Type:        schema.TypeString,
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Model parameters as a JSON string.", !isDataSource),
+			DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+				return jsonEqual(old, new)
+			},
+		},
+		CUSTOM_PARAMS: {
+			Type:        schema.TypeString,
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Custom model parameters as a JSON string.", !isDataSource),
+			DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+				return jsonEqual(old, new)
+			},
+		},
+		TAGS: tagsSchema(tagsSchemaOptions{isDataSource: isDataSource}),
+		COST_PER_INPUT_TOKEN: {
+			Type:        schema.TypeFloat,
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Cost per input token in USD.", !isDataSource),
+		},
+		COST_PER_OUTPUT_TOKEN: {
+			Type:        schema.TypeFloat,
+			Optional:    !isDataSource,
+			Computed:    isDataSource,
+			ForceNew:    !isDataSource,
+			Description: addForceNewDescription("Cost per output token in USD.", !isDataSource),
+		},
+		VERSION: {
+			Type:        schema.TypeInt,
+			Computed:    true,
+			Description: "Version of the AI model config.",
+		},
+		GLOBAL: {
+			Type:        schema.TypeBool,
+			Computed:    true,
```

Comment:
is this user-defined or does LD auto-assign the value? if the former it should not be computed

Reply:
LD auto-assigns